crates/blockchain/mempool.rs

```diff
@@ -206,6 +209,34 @@ fn validate_transaction(tx: &Transaction, store: Store) -> Result<(), MempoolErr
         }
     }
 
+    let maybe_sender_acc_info = store.get_account_info(header_no, tx.sender())?;
+
+    if let Some(sender_acc_info) = maybe_sender_acc_info {
+        if tx.nonce() < sender_acc_info.nonce {
+            return Err(MempoolError::InvalidNonce);
+        }
+
+        let effective_gas_price: U256 = tx
+            .effective_gas_price(block_header.base_fee_per_gas)
+            .ok_or(MempoolError::InvalidTxGasvalues)?
+            .into();
+        let gas_limit = U256::from(tx.gas_limit());
+        let tx_cost: U256 = effective_gas_price * gas_limit + tx.value();
+
+        if tx_cost > sender_acc_info.balance {
+            return Err(MempoolError::NotEnoughBalance);
+        }
```

Comment:
Wouldn't this reject transactions if the current block has a high base fee? Shouldn't we keep them in the pool and see if we can include them on a later block with a lower base fee?

Reply:
yeah that's correct, I changed it in this commit [672a3bf](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1125/commits/672a3bfe98d60d0ea060b11b685477d788871cd7) (I additionally switched to using the `max_fee_per_gas` field for 1559/4844 transactions, since that field is the actual maximum gas fee the user is willing to pay, irrespective of what the base and priority fees are.